### PR TITLE
[gitpod-db] Fix AuthProviderEntryDBImpl.findByUserId

### LINKED ISSUE #16412: Missing git integrations entries
### Problem to solve

Git integration entries are missing on Gitpod.io.

From the [relevant discussion](https://gitpod.slack.com/archives/C04QD5ZSKB2/p1676456671549019) (internal):

> @AlexTugarev: FYI, the missing Git Integrations are not subject to a DB table diff. There is a bug in server, which breaks the listing of providers. I'm on a fix. We'll need to push that to prod asap.

### PR DESCRIPTION
**Pull request by AlexTugarev**
## Related Issue(s)

Fixes #16412

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-slow-database
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Review by easyCZ — APPROVED** [review]

**Comment by AlexTugarev** [comment]
Thanks @easyCZ! Just adding a test before releasing the breaks on this.

File: components/gitpod-db/src/auth-provider-entry.spec.db.ts [hunk 1]
```diff
@@ -118,6 +118,18 @@ export class AuthProviderEntryDBSpec {
         expect(results).to.deep.contain(ap1);
         expect(results).to.deep.contain(ap2);
     }
+
+    @test public async findByUserId() {
```
Comment (easyCZ) [inline components/gitpod-db/src/auth-provider-entry.spec.db.ts:122]: Thank you! I was just about to write that a unit test would've cought this.

**Comment by AlexTugarev** [comment]
/hold cancel

```
[components/gitpod-db:dbtest]     ✓ findByUserId
```

**Pull request merged**